perses/app/setup_relative_calculation.py

```diff
@@ -1144,5 +1139,46 @@ def _validate_endstate_energies_for_htf(hybrid_topology_factory_dict: dict, topo
             validate_endstate_energies_point(current_htf, endstate=endstate, minimize=True)
 
 
+def _generate_parsed_yaml(setup_options, input_yaml_file_path):
+    """
+    Creates YAML file with parsed setup options in the working directory of the simulation.
+
+    It adds timestamp and ligands names information (old and new).
+
+    Parameters
+    ----------
+    setup_options: dict
+        Dictionary with perses setup options. Meant to be the returned dictionary from 
+        ``perses.app.setup_relative_calculation.getSetupOptions``,
+    input_yaml_file_path: str or Path object
+        Path to input yaml file with perses parameters
+
+    Returns
+    -------
+    out_yaml_path: str
+        String with the path to the generated parsed yaml file.
+    """
+    from openff.toolkit.topology import Molecule
+    # The parsed yaml file will live in the experiment directory to avoid race conditions with other experiments
+    yaml_path = Path(setup_options['trajectory_directory'])
+    yaml_name = Path(input_yaml_file_path).name  # extract name from input/template yaml file.
+    time = datetime.datetime.now().strftime("%Y-%m-%d-%H:%M:%S")
+    yaml_parse_name = f"perses-{time}-{yaml_name}"
+    # Add timestamp information
+    setup_options["timestamp"] = time
+    # Read input sdf file and save into list -- We don't check stereochemistry
+    ligands_list = Molecule.from_file(setup_options['ligand_file'], allow_undefined_stereo=True)
```

Comment:
I had to explicitly tell `openff-toolkit` to allow undefined stereochemistry for the test in `test_cli_resume_repex` to pass. Apparently we have undefined stereochemistry in the ligands for that test. I think this is okay, but I'm making the comment just in case this is something we have to solve in the test system.

Reply:
Not that I'm not touching this test, I just changed the `setup_relative_calculation.py` to extract names from the ligands file using the `openff-toolkit`.